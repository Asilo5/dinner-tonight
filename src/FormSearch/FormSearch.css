.search-recipe {
    font-size: 32px;
    border: none;
    background-color: transparent;
    border-bottom: 1px solid black;
    color: white;
}

.search-btn {
  font-size: 20px;
  border-radius: 10px;
  margin: 5px;
  padding: 10px;
  font-weight: bold;
  border: none;
}

.search-btn:hover {
  background: #ff3838;
  color: white;
}