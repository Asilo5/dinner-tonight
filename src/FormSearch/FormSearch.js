import React from 'react';


const FormSearch = (props) => {
 
    return(
        <form onSubmit={props.getRecipe}>
            <input 
               className='search-recipe'
               placeholder='Search for recipe here'
               type='text'
               name='recipeName' 
               />
            <button className='search-btn'>Search</button>
        </form>
    )
        
}


export default FormSearch;