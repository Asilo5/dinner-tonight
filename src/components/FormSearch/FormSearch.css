.search-recipe {
    font-size: 32px;
    border: none;
    background-color: transparent;
    border-bottom: 1px solid black;
    color: black;
    width: 40%;
    caret-color: #ff3838;
}

.search-btn {
  font-size: 20px;
  border-radius: 10px;
  margin: 5px;
  padding: 10px;
  font-weight: bold;
  border: none;
  cursor: pointer;
  background-color: transparent;
}

.search-btn:hover {
  background: #ff3838;
  color: white;
}