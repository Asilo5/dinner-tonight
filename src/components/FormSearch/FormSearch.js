import React, { Component } from 'react';
import './FormSearch.css'


class FormSearch extends Component {
    constructor() {
      super();
      this.state = { 
        ingredient: '' 
      }
    }

    handleChange = (e) => {
      this.setState({ [e.target.name] : e.target.value})
    }

    submitRecipe = (e) => {
      e.preventDefault();

      this.props.getRecipe(this.state.ingredient);

      this.emptyInputs();
    }

    emptyInputs = () => {
      this.setState({ ingredient : ''});
    }

    render() {
        return(
            <form>
                <input 
                   className='search-recipe'
                   placeholder='What ingredient do you fancy tonight?..'
                   type='text' 
                   name='ingredient'
                   value={this.state.ingredient}
                   onChange={(e) => this.handleChange(e)}
                   />
                <button className='search-btn' onClick={(e) => this.submitRecipe(e)}>Search</button>
            </form>
        )
    }
}

export default FormSearch;