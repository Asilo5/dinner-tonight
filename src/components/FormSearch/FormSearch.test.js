import React from 'react';
import { shallow } from 'enzyme';
import FormSearch from './FormSearch';

describe('FormSearch', () => {

    let wrapper;
    let mockGetRecipe = jest.fn();

    beforeEach(() => {
        wrapper = shallow(<FormSearch getRecipe={mockGetRecipe}/>);
    })

    it('should match snapshot with all data passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it.skip('should call handleChange when input changes', () => {
        expect(wrapper.find('input').props().value).toEqual('');
        const mockEvent = {
            target : {
                value: 'vegan'
            }
        };
        wrapper.find('input').simulate('change', mockEvent);
        expect(wrapper.find('input').props().value).toEqual('vegan');
    })

    it('should update state when handleChange is invoked', () => {
        const mockEvent = {
            target : {
               name: 'ingredient',
               value: 'vegan'
            }
        };

        const expected = 'vegan';
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state('ingredient')).toEqual(expected);
    })

    it('should call submitRecipe when button is clicked', () => {
        wrapper.instance().submitRecipe = jest.fn();
        wrapper.instance().forceUpdate();
        const mockEvent = {
            preventDefault: jest.fn()
        };
        wrapper.find('button').simulate('click', mockEvent);
        expect(wrapper.instance().submitRecipe).toHaveBeenCalledWith(mockEvent);
    })

    it('should call getRecipe and emptyInputs when submitRecipe is called', () => {
        const mockEvent = {
            preventDefault: jest.fn()
        };
        wrapper.instance().emptyInputs = jest.fn();
        wrapper.instance().submitRecipe(mockEvent);

        expect(mockGetRecipe).toHaveBeenCalled();
        expect(wrapper.instance().emptyInputs).toHaveBeenCalled();
    })

    it('should update state when emptyInputs is called', () => {
        wrapper.setState({
            ingredient : 'vegan'
        })
  
        wrapper.instance().emptyInputs();
  
        const expected = '';
  
        expect(wrapper.state('ingredient')).toEqual(expected);
    })
})