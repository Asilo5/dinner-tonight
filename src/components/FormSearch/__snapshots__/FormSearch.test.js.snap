// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch should match snapshot with all data passing in correctly 1`] = `
<form>
  <input
    className="search-recipe"
    name="ingredient"
    onChange={[Function]}
    placeholder="What ingredient do you fancy tonight?.."
    type="text"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
